### PR TITLE
Add experimental support for release testing

### PR DESCRIPTION
**Pull request by marccampbell**
This adds the first iteration of the `cluster` command (functionality is gated via an entitlement in replicated vendor portal accounts)

### CODE REVIEW

File: cli/cmd/cluster_create.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		Name:                   r.args.createClusterName,
+		KubernetesDistribution: r.args.createClusterKubernetesDistribution,
+		KubernetesVersion:      r.args.createClusterKubernetesVersion,
+		NodeCount:              r.args.createClusterNodeCount,
+		VCpus:                  r.args.createClusterVCpus,
+		MemoryMiB:              r.args.createClusterMemoryMiB,
+		TTL:                    r.args.createClusterTTL,
+	}
+	_, err := kotsRestClient.CreateCluster(opts)
+	if err == platformclient.ErrForbidden {
```
Comment (divolgin) [inline cli/cmd/cluster_create.go:46]: `errors.Cause(err)`?

**Review by divolgin — DISMISSED** [review]

File: cli/cmd/cluster.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+func (r *runners) InitClusterCommand(parent *cobra.Command) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "cluster",
+		Short: "Manage test clusters",
+		Long:  ``,
```
Comment (TyBrown) [inline cli/cmd/cluster.go:11]: Do we want to add:
```
Hidden: true,
```
until launch?

File: cli/cmd/runner.go [hunk 3]
```diff
@@ -164,4 +164,19 @@ type runnerArgs struct {
 	addRegistryTokenFromStdIn             bool
 
 	testRegistryImage string
+
+	createClusterName                   string
+	createClusterOSDistribution         string
+	createClusterOSVersion              string
```
Comment (TyBrown) [inline cli/cmd/runner.go:170]: Are we removing OS Distribution/Version?

File: cli/print/clusters.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/replicatedhq/replicated/pkg/types"
+)
+
+// TODO: implement a -o wide, and expose nodecount, vcpus and memory also?
+var clustersTmplSrc = `ID	NAME	OS DISTRO	OS VERSION	K8S DISTRO	K8S VERSION	STATUS	CREATED	EXPIRES
+{{ range . -}}
+{{ .ID }}	{{ .Name }}	{{ .OSDistribution }}	{{ .OSVersion }}	{{ .KubernetesDistribution}}	{{ .KubernetesVersion	}}	{{ .Status }}	{{ .CreatedAt}}	{{ .ExpiresAt }}
```
Comment (TyBrown) [inline cli/print/clusters.go:13]: OS Distro/Version

File: pkg/kotsclient/cluster_create.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import (
+	"net/http"
+
+	"github.com/replicatedhq/replicated/pkg/types"
+)
+
+type CreateClusterRequest struct {
+	Name                   string `json:"name"`
+	OSDistribution         string `json:"os_distribution"`
+	OSVersion              string `json:"os_version"`
```
Comment (TyBrown) [inline pkg/kotsclient/cluster_create.go:12]: OS Distro/Version?

File: pkg/types/cluster.go [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package types
+
+import "time"
+
+type Cluster struct {
+	ID                     string    `json:"id"`
+	Name                   string    `json:"name"`
+	OSDistribution         string    `json:"os_distribution"`
+	OSVersion              string    `json:"os_version"`
```
Comment (TyBrown) [inline pkg/types/cluster.go:9]: OS Distro/Version?

**Review by TyBrown — DISMISSED** [review]

**Review by TyBrown — APPROVED** [review]

**Review by divolgin — APPROVED** [review]

**Pull request merged**